tests/unit/plugins/test_authority.py

```diff
@@ -0,0 +1,148 @@
+    rrset_channel = asyncio.Queue()
+    fake_authority = authority.GCEAuthority(
+        authority_config, rrset_channel)
+    fake_authority.session = auth_client._session
+    return fake_authority
+
+
+@pytest.fixture
+def get_mock_client(mocker, get_gce_client):
+    def _create_fake(klass, *args, **kwargs):
+        client = get_gce_client(klass, *args, **kwargs)
+        fake_client = mocker.Mock(client)
+        return fake_client
+    return _create_fake
+
+
+@pytest.mark.asyncio
+async def test_builder_creates_proper_authority(mocker, authority_config):
+    gce_client = mocker.MagicMock(name='gce_client')
+    crm_client = mocker.MagicMock(name='crm_client')
+    auth_client = mocker.MagicMock(name='auth_client')
+    authority_mod_patch = 'gordon_janitor_gcp.clients.'
+    mocker.patch(authority_mod_patch + 'gcrm.GCRMClient', crm_client)
+    mocker.patch(authority_mod_patch + 'gce.GCEClient', gce_client)
+    mocker.patch(authority_mod_patch + 'auth.GAuthClient', auth_client)
+
+    rrset_channel = asyncio.Queue()
+    kwargs = {'other': 'kwargs'}
+    builder = authority.GCEAuthorityBuilder(
+        authority_config, rrset_channel, **kwargs)
+    session = builder.session
+    with contextlib.closing(session):
+        gce_authority = builder.build_authority()
+
+    auth_client_calls = [
+        mocker.call(
+            authority_config['keyfile'],
+            scopes=authority_config['scopes'],
+            session=session),
+    ] * 2
+
+    auth_client.assert_has_calls(auth_client_calls, any_order=True)
+    expected_metadata_blacklist = [
+        dict([pair]) for pair in authority_config['metadata_blacklist']
+    ]
+    gce_client.assert_called_once_with(
+        auth_client(),
+        session,
+        blacklisted_tags=authority_config['tag_blacklist'],
+        blacklisted_metadata=expected_metadata_blacklist)
+
+    assert crm_client() is gce_authority.crm_client
+    assert gce_client() is gce_authority.gce_client
+    assert rrset_channel is gce_authority.rrset_channel
+    assert authority_config is gce_authority.config
+
+
+@pytest.mark.asyncio
+async def test_builder_error_on_required_config(authority_config, caplog):
+    caplog.set_level(logging.ERROR)
+    del authority_config['keyfile']
+    rrset_channel = asyncio.Queue()
+    kwargs = {}
+
+    builder = authority.GCEAuthorityBuilder(
+        authority_config, rrset_channel, **kwargs)
+    session = builder.session
+    with pytest.raises(exceptions.GCPConfigError), contextlib.closing(session):
+        builder.build_authority()
+    assert 1 == len(caplog.records)
+
+
+@pytest.mark.asyncio
+async def test_start_publishes_msg_to_channel(mocker, authority_config,
+                                              get_gce_client, create_mock_coro):
+    instance_data = [{
+        'hostname': f'host-{i}',
+        'internal_ip': f'192.168.1.{i}',
+        'external_ip': f'1.1.1.{i}'
+    } for i in range(1, 4)]
```

Comment:
This is a little confusing to read, would you mind doing something like this:

```py
_data = {
    'hostname': 'host-{i}',
    'internal_ip': '192.168.1.{i}',
    'external_ip': '1.1.1.{i}'
}
instance_data = [_data.format(i=i) for i in range(1 ,4)]
```

Reply:
Thanks for pointing this out. I wasn't able to think of a way to keep the comprehension and format the data, so I exploded it out into a for-loop, which looks much clearer.